sklearn/pipeline.py

```diff
@@ -1232,6 +1232,12 @@ def n_features_in_(self):
         # X is passed to all transformers so we just delegate to the first one
         return self.transformer_list[0][1].n_features_in_
 
+    def __sklearn_is_fitted__(self):
+        # Delegate whether feature union was fitted
+        for _, transformer, _ in self._iter():
+            check_is_fitted(transformer)
+        return True
```

Comment:
the issue with this implementation is that we allow all transformers to be `None` in `FeatureUnion`. `fit`'s code is:

```python
        transformers = self._parallel_func(X, y, fit_params, _fit_one)
        if not transformers:
            # All transformers are None
            return self
        self._update_transformer_list(transformers)
        return self
```

wouldn't it be better here to add a private `_fitted` attribute in `fit`, and in this method to check for that?

If we want to also support pre-fitting transformers and then putting them in a `FeatureUnion`, we could do something like:

``` python
if getattr(self, '_fitted', None) or all_transformers_fitted:
    return True
```

Reply:
I think the current solution works because `_iter` does not yield dropped estimators:

https://github.com/scikit-learn/scikit-learn/blob/231dd7b03b105e1c177dddfc8e4c2b5532cf2fb6/sklearn/pipeline.py#L1052-L1057

(Also `FeatureUnion` no longer supports `None` and uses `"drop" instead. )

Reply:
@adrinjalali If all the transformers are None, it is stateless and is by definition fitted (imho). This is the approach that is also taken by `FunctionTransformer`:

```python
    def __sklearn_is_fitted__(self):
        """Return True since FunctionTransfomer is stateless."""
        return True
```

Instead of all transformers being None, I am fine returning true even if all transformers are stateless (like FunctionTransformer).

Reply:
Regarding `drop`, what would you like to test for it? As you know, transformers that are dropped are not yielded by `_iter`.

Reply:
@thomasjpfan Thanks. I didn't see your message while composing my responses.